spec/support/shared_examples/transform.rb

```diff
@@ -0,0 +1,808 @@
+shared_examples "a directory transformer" do
+  describe '#into' do
+    subject do
+      DirectoryDiff.transform(source_directory).into(target_directory, options)
+    end
+    let(:options) { {} }
+
+    context 'the current version is an empty directory' do
+      let(:current_directory) { [] }
+      let(:new_directory) do
+        [
+          ['Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil, 'foo'],
+          ['Matthew Johnston', 'matthew@envoy.com', '415-441-3232', nil],
+          ['Adolfo Builes', 'adolfo@envoy.com', nil, nil]
+        ]
+      end
+
+      it 'returns :insert ops for each row and passes through extra fields untouched' do
+        expect(subject).to eq([
+          [:insert, 'Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil, 'foo'],
+          [:insert, 'Matthew Johnston', 'matthew@envoy.com', '415-441-3232', nil],
+          [:insert, 'Adolfo Builes', 'adolfo@envoy.com', nil, nil]
+        ])
+      end
+    end
+
+    context 'the new version is an empty directory' do
+      let(:current_directory) do
+        [
+          ['Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil],
+          ['Matthew Johnston', 'matthew@envoy.com', '415-441-3232', nil]
+        ]
+      end
+      let(:new_directory) { [] }
+
+      it 'returns :delete ops' do
+        expect(subject).to eq([
+          [:delete, 'Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil],
+          [:delete, 'Matthew Johnston', 'matthew@envoy.com', '415-441-3232', nil]
+        ])
+      end
+    end
+
+    context 'both versions are empty' do
+      let(:current_directory) { [] }
+      let(:new_directory) { [] }
+
+      it 'returns no ops' do
+        expect(subject).to eq([])
+      end
+    end
+
+    context 'the current version is exactly the same as the new directory except for the extra field' do
+      let(:current_directory) do
+        [
+          ['Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil],
+          ['Matthew Johnston', 'matthew@envoy.com', '415-441-3232', nil]
+        ]
+      end
+      let(:new_directory) do
+        [
+          ['Matthew Johnston', 'matthew@envoy.com', '415-441-3232', nil, 'foo'],
+          ['Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil]
+        ]
+      end
+
+      it 'returns :noops ops' do # reordered to show order doesn't matter
```

Comment:
okay, not relevant since this will change in the next PR

Reply:
What it meant to show was the order the records appear in the CSV doesn't matter. What's dictating the order is the order in current directory, but always taking into account assistants.

Later PRs changes this behavior to be dictated by CSV order so that temp table output can be the same as in memory by ordering on csv row number.